### PR TITLE
Go To Line Number Option

### PR DESCRIPTION
**Pull request by grouma**
Add a dialog that accepts a line number. Upon enter the code will scroll to that position (or to the end if the number is larger than the max number of lines). Also wire up a keyboard shortcut to open the dialog.

UI:
Screen Shot 2021-04-14 at 2 10 14 PM

Closes https://github.com/flutter/devtools/issues/2844

### CODE REVIEW

**Comment by kenzieschmoll** [comment]
does this handle bad input like letters and or negative numbers?

**Comment by grouma** [comment]
> does this handle bad input like letters and or negative numbers?

The input is restricted so you can't even type non digit numbers.

File: packages/devtools_app/lib/src/debugger/codeview.dart [hunk 1]
```diff
@@ -808,12 +809,67 @@ class ScriptPopupMenuOption {
...
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          TextField(
+            autofocus: true,
+            onSubmitted: (value) {
+              if (value.isNotEmpty) {
+                Navigator.of(context, rootNavigator: true).pop('dialog');
+                final line = int.parse(value);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/debugger/codeview.dart:853]: we should fail gracefully if int.parse fails or if the value is a negative number.

File: packages/devtools_app/lib/src/debugger/codeview.dart [hunk 2]
```diff
@@ -808,12 +809,67 @@ class ScriptPopupMenuOption {
...
+                _debuggerController.showScriptLocation(
+                  ScriptLocation(
+                    _debuggerController.scriptLocation.value.scriptRef,
+                    location: SourcePosition(line: line, column: 0),
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/debugger/codeview.dart:857]: perhaps we should support `line:column`. Would be pretty easy to add and would match what IntelliJ supports for go to

File: packages/devtools_app/lib/src/debugger/codeview.dart [hunk 3]
```diff
@@ -808,12 +809,67 @@ class ScriptPopupMenuOption {
...
+                  ),
+                );
+              }
+            },
+            decoration: const InputDecoration(labelText: 'Line Number'),
+            keyboardType: TextInputType.number,
+            inputFormatters: <TextInputFormatter>[
+              FilteringTextInputFormatter.digitsOnly
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/debugger/codeview.dart:865]: I see this now - looks like this should handle bad input.

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/lib/src/debugger/codeview.dart (hunk 2 above)
Comment (grouma) [reply packages/devtools_app/lib/src/debugger/codeview.dart:857]: We only scroll the line into view we don't adjust your cursor so supporting column doesn't make sense. We'd have to rework the code view to support that.

File: packages/devtools_app/lib/src/debugger/codeview.dart (hunk 1 above)
Comment (grouma) [reply packages/devtools_app/lib/src/debugger/codeview.dart:853]: It can't be a negative number due to the limited input.

**Comment by kenzieschmoll** [comment]
FYI you're probably blocked on https://github.com/flutter/devtools/pull/2903

**Pull request merged**